### PR TITLE
feat: add effectsClass decorator

### PR DESCRIPTION
**Pull request by wumaolinmaoan**
## Summary by CodeRabbit

- **New Features**
	- Introduced dynamic class registration for various components (EffectComponent, CameraController, InteractComponent, ParticleSystem, SpriteComponent, TextComponent, VFXItem, ThreeSpriteComponent, and Model Components) using the `@effectsClass` decorator.
- **Refactor**
	- Streamlined plugin imports and registrations across multiple modules for better code organization and maintainability.
	- Updated deserializer logic to dynamically instantiate classes, enhancing flexibility and reducing complexity.
- **Chores**
	- Removed duplicate and unnecessary code, such as duplicate `serialize` function definitions and outdated constructor mappings, for a cleaner codebase.

### CODE REVIEW

File: packages/effects-core/src/plugins/camera/camera-controller-node.ts [hunk 1]
```diff
@@ -5,7 +5,10 @@ import type { Engine } from '../../engine';
 import type { ValueGetter } from '../../math';
 import { createValueGetter } from '../../math';
 import { Transform } from '../../transform';
+import { effectsClass } from '../../decorators';
+import { DataType } from '@galacean/effects-core';
```
Comment (yiiqii) [inline packages/effects-core/src/plugins/camera/camera-controller-node.ts:9]: 路径问题，同包下使用相对路径

File: packages/effects-core/src/plugins/interact/interact-item.ts [hunk 2]
```diff
@@ -10,11 +10,14 @@ import { InteractMesh } from './interact-mesh';
 import { RendererComponent } from '../../components';
 import type { DragEventType } from './interact-vfx-item';
 import type { Renderer } from '../../render';
+import { effectsClass } from '../../decorators';
+import { DataType } from '@galacean/effects-core';
```
Comment (yiiqii) [inline packages/effects-core/src/plugins/interact/interact-item.ts:14]: 同上

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: packages/effects-core/src/plugins/camera/camera-controller-node.ts (hunk 1 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/plugins/camera/camera-controller-node.ts]: 已修改

File: packages/effects-core/src/plugins/interact/interact-item.ts (hunk 2 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/plugins/interact/interact-item.ts]: 已修改

File: packages/effects-core/src/decorators.ts [hunk 3]
```diff
@@ -43,6 +44,19 @@ export function getMergedStore (target: any): any {
...
 }
 
+export function serialize (sourceName?: string) {
+  return generateSerializableMember(0, sourceName); // value member
+}
+
+export function effectsClass (className: any) {
+  return (target: any, context?: any) => {
+    if (effectsClassStore[className]) {
+      console.warn('Class ' + className + ' 重复注册');
```
Comment (yiiqii) [inline packages/effects-core/src/decorators.ts:54]: 重复注册是否应该 return 掉？

File: packages/effects-core/src/plugins/particle/particle-system.ts [hunk 4]
```diff
@@ -22,6 +22,8 @@ import type { TrailMeshProps } from './trail-mesh';
 import { TimelineComponent } from '../cal/calculate-item';
 import { Track } from '../cal/track';
 import { ParticleBehaviourPlayable } from './particle-vfx-item';
+import { effectsClass } from '../../decorators';
+import { DataType } from '@galacean/effects-core';
```
Comment (yiiqii) [inline packages/effects-core/src/plugins/particle/particle-system.ts:26]: 路径问题

File: packages/effects-core/src/plugins/sprite/sprite-item.ts [hunk 5]
```diff
@@ -20,6 +20,8 @@ import { Track } from '../cal/track';
 import type { BoundingBoxTriangle, HitTestTriangleParams } from '../interact/click-handler';
 import { HitTestType } from '../interact/click-handler';
 import { getImageItemRenderInfo, maxSpriteMeshItemCount, spriteMeshShaderFromRenderInfo } from './sprite-mesh';
+import { effectsClass } from '../../decorators';
+import { DataType } from '@galacean/effects-core';
```
Comment (yiiqii) [inline packages/effects-core/src/plugins/sprite/sprite-item.ts:24]: 同上

File: packages/effects-core/src/plugins/text/text-item.ts [hunk 6]
```diff
@@ -7,6 +7,8 @@ import { TextLayout } from './text-layout';
 import { TextStyle } from './text-style';
 import { DEFAULT_FONTS, canvasPool } from '../../template-image';
 import { glContext } from '../../gl';
+import { effectsClass } from '../../decorators';
+import { DataType } from '@galacean/effects-core';
```
Comment (yiiqii) [inline packages/effects-core/src/plugins/text/text-item.ts:11]: 同上

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: packages/effects-core/src/decorators.ts (hunk 3 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/decorators.ts:54]: 现在 three 是通过覆盖注册来渲染组件的，后面看能不能修改 json 的 datatype 避免重复注册，这边就可以 return 了。

Comment (yiiqii) [reply packages/effects-core/src/decorators.ts:54]: 那加上 TODO 吧，避免忘记

File: packages/effects-core/src/plugins/particle/particle-system.ts (hunk 4 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/plugins/particle/particle-system.ts]: 已修改

File: packages/effects-core/src/plugins/sprite/sprite-item.ts (hunk 5 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/plugins/sprite/sprite-item.ts]: 已修改

File: packages/effects-core/src/plugins/text/text-item.ts (hunk 6 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/plugins/text/text-item.ts]: 已修改

**Review by yiiqii — APPROVED** [review]

**Pull request merged**